File: _BHCVRP_Python_Version/cujae_inf_citi_om/exceptions/VehicleSpeedException.py
class VehicleSpeedException(Exception):
    def __init__(self, vehicle_speed=None):
        if vehicle_speed is not None:
            self.vehicle_speed = vehicle_speed

    def __init__(self, message=None):
        if message is not None:
            super().__init__(message)

    def get_vehicle_speed(self):
        return getattr(self, "vehicle_speed", None)

File: BHCVRP_Java_Version/src/cujae/inf/citi/om/exceptions/VehicleSpeedException.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package cujae.inf.citi.om.exceptions;

/**
 *
 * @author kmych
 */
public class VehicleSpeedException extends Exception{
    private float vehicleSpeed;
    
    public VehicleSpeedException(float vehicleSpeed){
        this.vehicleSpeed = vehicleSpeed;
    
    }
    
    public float getVehicleSpeed(){
        return vehicleSpeed;
    }
    
    public VehicleSpeedException(String message){   
        super(message);
    }
    
}